xla/service/gpu/fusions/mlir/mlir_fusion_emitter.cc

```diff
@@ -590,8 +593,21 @@ void AddLoweringPasses(mlir::OpPassManager& pm,
   pm.addPass(mlir::createLoopInvariantCodeMotionPass());
   pm.addPass(mlir::createSymbolDCEPass());
   pm.addPass(mlir::createCSEPass());
-  pm.addPass(CreateExpandFloatOpsPass(
-      !device.cuda_compute_capability().IsAtLeastAmpere()));
+
+  auto ptx_version =
```

Comment:
We need to guard this call with an ifdef and only add the include if we're compiling for Nvidia. See MinThreadsXRowReduction in xla/service/gpu/reduction_utils.cc for an example. That function also says that this function is expensive, so we will probably want to have some caching here as well (or maybe unify that somehow, but that doesn't need to be in this PR).

Reply:
A few lines above in the same file (line 578) we access `device.cuda_compute_capability()` without an `#ifdef`
What is the downside of this approach, compared to using an `#ifdef GOOGLE_CUDA` ? (btw, updated the PR)

Reply:
`cuda_compute_capability` exists in both builds, but `cuda_asm_compiler` isn't built in a ROCm build, so it needs to be guarded by an `ifdef` in the cc file and an `if_cuda_is_configured` in the BUILD file. We could definitely centralize these guards and the caching, but maybe not in this PR?

Reply:
Clear, thanks. Updated the PR.

Reply:
If the comment in reduction_utils is true, we still need caching here, or maybe better directly in se::GetAsmCompilerVersion.

Reply:
After reading the code in reduction_utils, I still cannot figure out why atomics are needed there.

Anyway, added a simple utility for getting a value once during the execution (haven't found a similar one in the XLA codebase) - please suggest a namespace where I should put it in (it doesn't really belong to stream_executor).

Reply:
Looks good in principle, but I think if we want to use this, we have to do it directly in `GetAsmCompilerVersion`, since otherwise the value won't be cached across multiple invocations in different code locations (we don't want to evaluate it once from `reduction_utils` and then again from here).

As to where to put it, I can't really find a really good place either. How about xla/util.h? That contains a bunch of semi-related utilities, so at least we're not making it much worse.

Reply:
Moved the helper to `util.h` and updated `GetAsmCompilerVersion`.

We can also update a similar method `GetNvLinkVersion` and remove the redundancy in `reduction_utils` (in a separate PR).